### DISCUSSION TITLE
Set an specific size to a non floating window

### DISCUSSION THREAD

**Discussion by victor-falcon** [asmvik/yabai#2142] [Q&A] [answered]
I have a second display where I like to keep two windows with this ratio.

![image](https://github.com/koekeishiya/yabai/assets/238766/56731213-b2d1-49c6-b8de-8e307c0cfd15)

RIght now I see myself resizing windows manually, Is there any way to set a shortcut to set this ratio automatically?

**Comment by tjex** [comment]
the tiling is strictly set to binary partitioning, there are no custom ratios from what I understand. 
If you're consistently using specific apps / widows in that configuration you could look at setting a `rule` so that those apps / winow titles get placed there as floating widnows, but on a specific monitor and in a specific position:

see here: https://github.com/koekeishiya/yabai/wiki/Commands#move-window

**Comment by tjex** [comment]
I remembered there's the split ratio option: https://github.com/koekeishiya/yabai/wiki/Configuration#split-ratios

But this will apply universally to all screens / spaces, etc. Perhaps it's still of interest

**Comment by Bellavene** [comment]
You need to write a script for ratios per monitor

**Reply by victor-falcon** [reply]
Do you have an example of an script to resize a window until the height eq X?

**Reply by victor-falcon** [reply]
I have this right now, but now working 😞

```
#!/bin/bash

TARGET_HEIGHT=910
WINDOW_RESIZE_AMOUNT=10

while [[ $(yabai -m query --windows --window | jq '.frame."h"' | awk '{print int($1)}') -lt $TARGET_HEIGHT ]]; do
	yabai -m window --resize bottom:0:-$WINDOW_RESIZE_AMOUNT
done
```

**Reply by Bellavene** [reply]
For some reason I had to add `window_placement last_child` to `$focus == 2` to make it to work properly, as I have usually set it to first_child, otherwise it applies split ratio only when third window is added... Who knows why. Ideally you need to add windows count, to allow split ratio change only when a second window is created on your display.
Have fun.

.yabairc
```
yabai -m signal --add event=display_changed action="yabai-per-display"
yabai -m signal --add event=space_changed action="yabai-per-display"
```

yabai-per-display
```
#!/bin/zsh
disp=$(yabai -m query --spaces --display)
focus=$(echo $disp | jq '.[0]."index"')

if [[ $focus == "2" ]]; then
	yabai -m config \
	split_type horizontal \
	split_ratio 0.7 \
	window_placement last_child
else
	yabai -m config \
	split_type auto \
	split_ratio 0.5
fi
```

**Reply by victor-falcon** [reply, marked as answer]
Finally I went with this script and map it to a shortcut.

```
#!/bin/bash

TARGET_HEIGHT=890
WINDOW_HEIGHT=$(yabai -m query --windows --window | jq '.frame."h"')
WINDOW_HEIGHT=$(echo $WINDOW_HEIGHT | tr -d '"')
RESIZE_AMOUNT=$(echo "$WINDOW_HEIGHT - $TARGET_HEIGHT" | bc)

if (($(echo "$RESIZE_AMOUNT > 0" | bc -l))); then
	yabai -m window --resize bottom:0:-$RESIZE_AMOUNT
fi
```
